core/model/modx/rest/modrestservice.class.php

```diff
@@ -395,32 +395,45 @@ public function setRequestParameters() {
      * @return array
      */
 	protected function _collectRequestParameters() {
-	    $data = @file_get_contents('php://input');
-	    $params = array();
-	    $contentType = isset($_SERVER['CONTENT_TYPE']) ? $_SERVER['CONTENT_TYPE'] : '';
-	    $spPos = strpos($contentType,';');
-	    if ($spPos !== false) {
-	        $contentType = substr($contentType,0,$spPos);
-	    }
+        $filehandle = fopen('php://input', "r");
+        $params = array();
+        $contentType = isset($_SERVER['CONTENT_TYPE']) ? $_SERVER['CONTENT_TYPE'] : '';
+        $spPos = strpos($contentType, ';');
+        if ($spPos !== false) {
+            $contentType = substr($contentType, 0, $spPos);
+        }
         switch ($contentType) {
+            case 'image/jpeg':
+            case 'image/png':
+            case 'image/gif':
+                $params['filehandle'] = $filehandle;
+                break;
             case 'application/xml':
             case 'text/xml':
+                $data = stream_get_contents($filehandle);
+                fclose($filehandle);
                 $xml = simplexml_load_string($data);
-                $params = $this->_xml2Array($xml);
+                $params = $this->_xml2array($xml);
                 break;
             case 'application/json':
             case 'text/json':
+                $data = stream_get_contents($filehandle);
+                fclose($filehandle);
                 $params = $this->service->modx->fromJSON($data);
                 break;
             case 'application/x-www-form-urlencoded':
             default:
+                $data = stream_get_contents($filehandle);
+                fclose($filehandle);
                 parse_str($data, $params);
                 break;
         }
-        if ($this->service->getOption('trimParameters',false)) {
-		    array_walk_recursive($this->parameters,array('modRestServiceRequest','_trimString'));
+        if ($this->service->getOption('trimParameters', false)) {
+            array_walk_recursive($this->parameters, array('modRestServiceRequest', '_trimString'));
+        }
+        if (empty($params)) {
```

Comment:
I have not changed the code on that part. It is only reformatted with curly braces.

$params could get null in line 422 with an invalid JSON, so the check makes sense. But it could be changed to !is_array() for sure.